src/types.ts

```diff
@@ -3488,6 +3488,122 @@ export class ZodNaN extends ZodType<number, ZodNaNDef> {
+//////////      ZodFile         //////////
+//////////                     //////////
+/////////////////////////////////////////
+/////////////////////////////////////////
+
+type ZodFileCheck =
+  | { kind: "max"; value: number; message?: string }
+  | { kind: "min"; value: number; message?: string }
+  | { kind: "type"; value: string | string[]; message?: string };
+
+export interface ZodFileDef extends ZodTypeDef {
+  typeName: ZodFirstPartyTypeKind.ZodFile;
+  checks: ZodFileCheck[];
+}
+
+export class ZodFile extends ZodType<File, ZodFileDef> {
+  _parse(input: ParseInput): ParseReturnType<File> {
+    const { status, ctx } = this._processInputParams(input);
+
+    if (!(input.data instanceof File)) {
+      addIssueToContext(ctx, {
+        code: ZodIssueCode.invalid_type,
+        expected: ZodParsedType.file,
+        received: ctx.parsedType,
+      });
+
+      return INVALID;
+    }
+
+    for (const check of this._def.checks) {
+      if (check.kind === "max") {
+        if (ctx.data.size > check.value) {
+          addIssueToContext(ctx, {
+            code: ZodIssueCode.too_big,
+            maximum: check.value,
+            type: "file",
+            inclusive: true,
+            message: check.message,
+          });
+          status.dirty();
+        }
+      } else if (check.kind === "min") {
+        if (ctx.data.size < check.value) {
+          addIssueToContext(ctx, {
+            code: ZodIssueCode.too_small,
+            minimum: check.value,
+            type: "file",
+            inclusive: true,
+            message: check.message,
+          });
+          status.dirty();
+        }
+      } else if (check.kind === "type") {
+        const types = Array.isArray(check.value) ? check.value : [check.value];
+
+        if (!types.includes(ctx.data.type)) {
+          addIssueToContext(ctx, {
+            code: ZodIssueCode.invalid_file_type,
+            expected: types,
+            received: ctx.data.type,
+            message: check.message,
+          });
+          status.dirty();
+        }
+      }
+    }
+
+    return { status: status.value, value: ctx.data };
+  }
+
+  _addCheck(check: ZodFileCheck) {
+    return new ZodFile({
+      ...this._def,
+      checks: [...this._def.checks, check],
+    });
+  }
+
+  max(maxSize: number, message?: errorUtil.ErrMessage) {
+    return this._addCheck({
+      kind: "max",
+      value: maxSize,
+      ...errorUtil.errToObj(message),
+    });
+  }
+
+  min(minSize: number, message?: errorUtil.ErrMessage) {
+    return this._addCheck({
+      kind: "min",
+      value: minSize,
+      ...errorUtil.errToObj(message),
+    });
+  }
+
+  type(accept: string | string[], message?: errorUtil.ErrMessage) {
+    return this._addCheck({
+      kind: "type",
+      value: accept,
+      ...errorUtil.errToObj(message),
+    });
+  }
+
+  static create = (): ZodFile => {
+    if (typeof File === "undefined") {
+      throw new Error("File not supported in current environment");
```

Comment:
We could use something like [@web-std/file](https://github.com/web-std/io/tree/main/packages/file)